pkg/tsdb/azuremonitor/azure-log-analytics-datasource.go

```diff
@@ -135,7 +135,8 @@ func (e *AzureLogAnalyticsDatasource) executeQuery(ctx context.Context, query *A
 		return dataResponse
 	}
 
-	if !dsInfo.Settings.AzureLogAnalyticsSameAs {
+	// If azureLogAnalyticsSameAs is defined and set to false, return an error
+	if sameAs, ok := dsInfo.JSONData["azureLogAnalyticsSameAs"]; ok && !sameAs.(bool) {
```

Comment:
Are we able to (and what do you think of) breaking this out into two variables to make a bit more readable, something like

```go
sameAsConfigured := dsInfo.JSONData["azureLogAnalyticsSameAs"];

if sameAsConfigured && !sameAs {
  ....
}
```

Reply:
the `ok` variable doesn't mean that the value is `true`, it means that the entry `azureLogAnalyticsSameAs` exists in the map (the variable name is a golang idiom): https://blog.golang.org/maps

What is really getting the value is `sameAs.(bool)`. I need to convert it to a `bool` because the map is defined as `map[string]interface{}` being `interface{}` a generic type. :)